src/Sulu/Bundle/CategoryBundle/Resources/public/js/validation/types/categoryList.js

```diff
@@ -38,8 +38,11 @@ define([
                     App.emit('sulu.content.changed');
                 },
 
-                setValue: function(value) {
-                    App.dom.data($el, 'selected', value);
+                setValue: function(categories) {
+                    categories = categories.map(function(category) {
+                        return (!!category.id) ? category.id : category;
```

Comment:
Same question as before (I think you only moved the problem to somewhere else): Why do you get different values here?